### ISSUE TITLE
How to train on my computer

### ISSUE THREAD

**Issue by liguangzi** [LZQthePlane/Online-Realtime-Action-Recognition-based-on-OpenPose#12]
I want to train a model to recognize the stand walk operation fall_down four actions in the video. Would I open train.py and then modify the following three sentences: 

1. # model. save ('framewise_recognition. h5') - model. save ('framewise_recognition. h5')

2. model. fit (X_train, Y_train, batch_size=32, epochs=20, verbose=1, validation_data= (X_test, Y_test))-

Model. fit (X_train, Y_train, batch_size=32, epochs=20, verbose=1, validation_data= (X_test, Y_test), callbacks= [his])

3. raw_data = pd. read_csv ('data_with_scene. csv', header = 0)-

Raw_data = pd. read_csv ('data_under_scene. csv', header = 0)

Is that the case?

**Comment by LZQthePlane** [comment]
yes

**Comment by liguangzi** [comment]
But why is the training time only tens of seconds, the training process should not be very long, I am a novice

Running train.py results in this:

Use Tensor Flow backend.

Train on 2960 samples, validate on 329 samples

Epoch 1/20

2019-03-21 14:21:40.189386: I T: src github tensorflow tensorflow tensorflow core platform feature_guard.cc:141] Your CPU support instructions that Tensor Flow binary was not compiled to use: AVX2



32/2960 [.....................] - ETA: 2:07 - loss: 2.2246 - acc: 0.1875

640/2960 [====>............] - ETA: 5S - loss: 1.7659 - acc: 0.2969

1344/2960 [===================>......] - ETA: 1s - loss: 1.4906 - acc: 0.4033

1984/2960 [================================> etc.] - ETA: 0s - loss: 1.3425 - acc: 0.4738

2592/2960 [=========================================>...] - ETA: 0s - loss: 1.2270 - acc: 0.5216

2960/2960 [====================================================] - 2S 584us/step-loss: 1.1752 - acc: 0.5443 - val_loss: 0.7969 - val_acc: 0.7447

Epoch 2/20



32/2960 [.....................] - ETA: 0s - loss: 0.7450 - acc: 0.7812

704/2960 [=====>..................] - ETA: 0s - loss: 0.7356 - acc: 0.7401

1312/2960 [===================>......] - ETA: 0s - loss: 0.7096 - acc: 0.7652

1856/2960 [=============================> etc.] - ETA: 0s - loss: 0.6678 - acc: 0.7926

2496/2960 [=======================================>...] - ETA: 0s - loss: 0.6358 - acc: 0.8133

2960/2960 [=====================================================] - 0s 87us/step-loss: 0.6196 - acc: 0.8226 - val_loss: 0.4971 - val_acc: 0.8784

Epoch 3/20



32/2960 [.....................] - ETA: 0s - loss: 0.4597 - acc: 0.8750

704/2960 [=====>..................] - ETA: 0s - loss: 0.4812 - acc: 0.9020

1280/2960 [==================>...] - ETA: 0s - loss: 0.4660 - acc: 0.9039

1888/2960 [==============================> etc.] - ETA: 0s - loss: 0.4506 - acc: 0.9089

2464/2960 [====================================> etc] - ETA: 0s - loss: 0.4399 - acc: 0.9107

2960/2960 [====================================================] - 0s 86us/step-loss: 0.4286 - acc: 0.9132 - val_loss: 0.3689 - val_acc: 0.9422

Epoch 4/20



32/2960 [.....................] - ETA: 0s - loss: 0.2771 - acc: 0.9375

672/2960 [====>............] - ETA: 0s - loss: 0.3472 - acc: 0.9345

1312/2960 [===================>......] - ETA: 0s - loss: 0.3343 - acc: 0.9383

1952/2960 [==============================> etc.] - ETA: 0s - loss: 0.3247 - acc: 0.9426

2592/2960 [========================================>...] - ETA: 0s - loss: 0.3208 - acc: 0.9429

2960/2960 [===================================================] - 0s 90us/step-loss: 0.3158 - acc: 0.9443 - val_loss: 0.2785 - val_acc: 0.9453

Epoch 5/20



32/2960 [.....................] - ETA: 0s - loss: 0.2925 - acc: 0.8750

480/2960 [==>..................] - ETA: 0s - loss: 0.2901 - acc: 0.9458

960/2960 [=========>.........] - ETA: 0s - loss: 0.2726 - acc: 0.9563

1440/2960 [=====================>......] - ETA: 0s - loss: 0.2639 - acc: 0.9618

1888/2960 [===============================> etc.] - ETA: 0s - loss: 0.2687 - acc: 0.9560

2336/2960 [==================================> etc.] - ETA: 0s - loss: 0.2607 - acc: 0.9619

2816/2960 [===============================================>.] - ETA: 0s - loss: 0.2552 - acc: 0.9631

2960/2960 [==================================================] - 0s 115us/step-loss: 0.2544 - acc: 0.9639 - val_loss: 0.2230 - val_acc: 0.9574

Epoch 6/20



32/2960 [.....................] - ETA: 0s - loss: 0.1596 - acc: 0.9688

480/2960 [==>..................] - ETA: 0s - loss: 0.2551 - acc: 0.9521

896/2960 [=========>.........] - ETA: 0s - loss: 0.2439 - acc: 0.9621

1408/2960 [=====================> etc.] - ETA: 0s - loss: 0.2263 - acc: 0.9680

2048/2960 [================================> etc.] - ETA: 0s - loss: 0.2138 - acc: 0.9712

2720/2960 [===========================================>...] - ETA: 0s - loss: 0.2106 - acc: 0.9710

2960/2960 [====================================================] - 0s 98us/step-loss: 0.2092 - acc: 0.9706 - val_loss: 0.1833 - val_acc: 0.9605

Epoch 7/20



32/2960 [.....................] - ETA: 0s - loss: 0.1706 - acc: 1.0000

608/2960 [====>............] - ETA: 0s - loss: 0.1819 - acc: 0.9770

1280/2960 [==================>...] - ETA: 0s - loss: 0.1934 - acc: 0.9648

1952/2960 [===============================> etc.] - ETA: 0s - loss: 0.1851 - acc: 0.9713

2560/2960 [======================================>...] - ETA: 0s - loss: 0.1784 - acc: 0.9730

2960/2960 [====================================================] - 0s 88us/step-loss: 0.1767 - acc: 0.9747 - val_loss: 0.1483 - val_acc: 0.9696

Epoch 8/20



32/2960 [.....................] - ETA: 0s - loss: 0.1393 - acc: 1.0000

608/2960 [====>............] - ETA: 0s - loss: 0.1427 - acc: 0.9836

1248/2960 [==================>...] - ETA: 0s - loss: 0.1541 - acc: 0.9800

1952/2960 [===============================> etc.] - ETA: 0s - loss: 0.1516 - acc: 0.9805

2528/2960 [=======================================>...] - ETA: 0s - loss: 0.1516 - acc: 0.9818

2960/2960 [======================================================] - 0s 87us/step-loss: 0.1496 - acc: 0.9828 - VA
Is that the case?

**Comment by LZQthePlane** [comment]
It's a small dataset, training just takes ten of seconds.

**Comment by liguangzi** [comment]
Okay, thank you. but why do we only have walk and fall-down tags when we use the model trained with this small training data set? When people are standing and sitting, the tags are not recognized.

**Comment by LZQthePlane** [comment]
my model of action recognizing was trained and test in a specific environment, which means the resolution/positions/angles...of the camera and person were designed. so if your trial of test is not just as same as mine, it may result in problems.
 the best way is retrained the model with your own dataset, collecting the data in your specific environment. My work is just a pipline of experiment. Hope it will help you.

**Comment by liguangzi** [comment]
Okay, thank you. So how do I use my own dataset? I need to recognize the basic movements of the characters in the video (sitting, running, walking, falling). Where should I download them? Can you show me the way?

**Comment by LZQthePlane** [comment]
I have showed in my `README.md`

**Comment by liguangzi** [comment]
Well, thank you very much. I'll study it first.
